x-pack/plugins/viz_editor/common/query_types.ts

```diff
@@ -114,8 +114,6 @@ export interface OrderByOperation {
 }
 
 export interface Query {
-  index: string;
```

Comment:
You removed `index` here but didn't add `datasourceId`?

Reply:
When there is just one datasource at a time, we don't need an id. But you are right, we could handle multi-index with that - there could be a separate datasource plugin "multi-indexpattern" (possibly in x-pack) which supports selecting fields from different indexpatterns and the `datasourceId` in the query is something which is picked up by the `toExpression` function of the datasource plugin to translate it to the indexpattern name